client/src/Movies/Movie.js

```diff
@@ -18,7 +19,7 @@ const Movie = (props) => {
           console.error(error);
         });
 
-  },[]);
+  },[props]);
```

Comment:
In the second argument we should include the array which useEffect depends on, otherwise, it will fire any time the component is re-rendered. Here is a great resource for more on that: https://reactjs.org/docs/hooks-reference.html#conditionally-firing-an-effect